rimage/image_file.go

```diff
@@ -34,6 +36,41 @@ var RGBABitmapMagicNumber = []byte("RGBA")
 // in bytes. See above as to why.
 const RawRGBAHeaderLength = 12
 
+func init() {
+	// Here we register the custom format above so that we can simply use image.Decode
+	// so long as the raw RGBA data has the appropriate header
+	image.RegisterFormat("vnd.viam.rgba", string(RGBABitmapMagicNumber),
+		func(r io.Reader) (image.Image, error) {
+			rawBytes, err := ioutil.ReadAll(r)
+			if err != nil {
+				return nil, err
+			}
+			header := rawBytes[:RawRGBAHeaderLength]
+			width := int(binary.BigEndian.Uint32(header[4:8]))
+			height := int(binary.BigEndian.Uint32(header[8:12]))
+			img := image.NewNRGBA(image.Rect(0, 0, width, height))
+			imgBytes := rawBytes[RawRGBAHeaderLength:]
+			img.Pix = imgBytes
+			return img, nil
+		},
+		func(r io.Reader) (image.Config, error) {
+			imgBytes := make([]byte, RawRGBAHeaderLength)
+			_, err := r.Read(imgBytes)
+			if err != nil {
+				return image.Config{}, err
+			}
+			header := imgBytes[:RawRGBAHeaderLength]
```

Comment:
Shouldn't the imageBytes already be as long as the Length? Though you may also want to check for a panic here as well if it didn't end up reading enough bytes.